post-processor/vsphere-template/step_mark_as_template.go

```diff
@@ -51,42 +52,56 @@ func (s *stepMarkAsTemplate) Run(ctx context.Context, state multistep.StateBag)
 		return multistep.ActionHalt
 	}
 
+	// Use a simple "MarkAsTemplate" method unless `reregister_vm` is true
+	if !s.ReregisterVM {
+		ui.Message("Marking as a template...")
+
+		if err := vm.MarkAsTemplate(context.Background()); err != nil {
+			state.Put("error", err)
+			ui.Error("vm.MarkAsTemplate:" + err.Error())
+			return multistep.ActionHalt
+		}
+		return multistep.ActionContinue
+	}
+
+	ui.Message("Re-register VM as a template...")
+
 	dsPath, err := datastorePath(vm)
 	if err != nil {
 		state.Put("error", err)
-		ui.Error(err.Error())
+		ui.Error("datastorePath:" + err.Error())
 		return multistep.ActionHalt
 	}
 
 	host, err := vm.HostSystem(context.Background())
 	if err != nil {
 		state.Put("error", err)
-		ui.Error(err.Error())
+		ui.Error("vm.HostSystem:" + err.Error())
 		return multistep.ActionHalt
 	}
 
 	if err := vm.Unregister(context.Background()); err != nil {
 		state.Put("error", err)
-		ui.Error(err.Error())
+		ui.Error("vm.MarkAsTemplate:" + err.Error())
```

Comment:
s/vm.MarkAsTemplate/vm.Unregister/g

Reply:
Ooops! Copy/paste got me, I fixed this as an "amend" to prevent a bunch of typo commits.